src/components/App/ImportStudyList.js

```diff
@@ -0,0 +1,168 @@
+import React, { PureComponent } from 'react';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    TextField,
+} from '@material-ui/core';
+import { queryWebsoc } from '../../helpers';
+import RightPaneStore from '../../stores/RightPaneStore';
+import { addCourse, openSnackbar } from '../../actions/AppStoreActions';
+import AppStore from '../../stores/AppStore';
+import { GetApp } from '@material-ui/icons';
+import { termData } from '../../termData';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+
+class ImportStudyList extends PureComponent {
+    state = {
+        isOpen: false,
+        selectedTerm: RightPaneStore.getFormData().term,
+        studyListText: '',
+    };
+
+    handleChange = (event) => {
+        this.setState({ selectedTerm: event.target.value });
+    };
+
+    handleError = (error) => {
+        openSnackbar('error', 'An error occurred while trying to import the Study List.');
+        console.error(error);
+    };
+
+    handleOpen = () => {
+        this.setState({ isOpen: true });
+    };
+
+    handleClose = async (doImport) => {
+        this.setState({ isOpen: false }, () => {
+            document.removeEventListener('keydown', this.enterEvent, false);
+            this.setState({ studyListText: '' });
+        });
+        if (doImport) {
+            document.removeEventListener('keydown', this.enterEvent, false);
+            if (!this.state.studyListText.match(/\d{5}/g)) {
+                openSnackbar('error', 'Cannot import an empty/invalid Study List.');
+                return;
+            }
+            const currSchedule = AppStore.getCurrentScheduleIndex();
+            const sectionCodes = this.state.studyListText
+                .split('\n')
+                .map((line) => line.match(/\d{5}/g))
```

Comment:
You can do this matching without splitting the string and mapping each line. Running match on the entire study list string should do the trick. See this screenshot:
![(running a regex match in the console to illustrate the point)](https://user-images.githubusercontent.com/48658337/158674972-acd3b4b5-c3e5-497e-a125-341d54ecbd51.png)
Also I think the filter is unnecessary, and also if you run the match on the entire string isntead of each line the flatten shouldn't be needed either.